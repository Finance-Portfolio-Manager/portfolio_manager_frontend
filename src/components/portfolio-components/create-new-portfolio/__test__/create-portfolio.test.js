import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import NewPortfolio from './../create-portfolio';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
const wrapper = mount(<Router><NewPortfolio></NewPortfolio></Router>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<Router><NewPortfolio></NewPortfolio></Router>).toJSON();
    expect(tree).toMatchSnapshot();
})

test("simulate onChange for first input", ()=>{
    const input = wrapper.find("input").first();
    const changeAction = ()=>{
        input.simulate("change", {target: {value: true}});
    }
    expect(changeAction).not.toThrow();
})

test("simulate click for submit button", ()=>{
    const submitButton = wrapper.find("button");
    const clickAction = ()=>{
        submitButton.simulate('click');
    }
    expect(clickAction).not.toThrow();
})