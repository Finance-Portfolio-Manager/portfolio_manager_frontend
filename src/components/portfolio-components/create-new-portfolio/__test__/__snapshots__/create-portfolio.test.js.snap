// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div>
    <label
      className="form-label"
    >
      Portfolio Name
    </label>
    <input
      className="form-control"
      name="name"
      onChange={[Function]}
    />
    <div
      className="form-check"
    >
      <input
        className="form-check-input"
        disabled={false}
        name="isPublic"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="form-check-label"
        title=""
      >
        Public
      </label>
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
     Submit 
  </button>
</form>
`;
